jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/ResultAttribute.java

```
@@ -0,0 +1,74 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.model;
+
+import java.lang.reflect.Method;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Enumerates the attributes of {@link Result}
+ * 
+ * @author Simon Hutchinson
+ *         <a href="https://github.com/sihutch">github.com/sihutch</a>
+ *
+ */
+public enum ResultAttribute {
+
+	TYPENAME("typeName"), OBJDOMAIN("objDomain"), CLASSNAME("className"), ATTRIBUTENAME("attributeName");
+
+	private String attributeName;
+	private String accessorMethod;
+
+	private ResultAttribute(String attributeName) {
+		this.attributeName = attributeName;
+		this.accessorMethod = "get" + StringUtils.capitalize(attributeName);
+	}
+
+	public String getAttributeName() {
+		return attributeName;
+	}
+
+	/**
+	 * Calls the Getter defined by the {@link ResultAttribute} on the {@link Result}
+	 * add adds the entry to the supplied {@link Map}
+	 * 
+	 * @param attributeMap
+	 *            The map to add the {@link Result} data to
+	 * @param result
+	 *            The {@link Result} to get the data from
+	 * @throws ReflectionException
+	 *             If reflection cannot be performed on the {@link Result}
+	 */
+	public void addAttribute(Map<String, String> attributeMap, Result result) throws ReflectionException {
+		if (attributeMap != null && result != null) {
+			try {
+				Method m = result.getClass().getMethod(accessorMethod);
+				attributeMap.put(attributeName, (String) m.invoke(result));
+			} catch (Exception e) {
```

Wrapping the exceptions here does not bring much additional value. I'd prefer to let them bubble up, adding `java.lang.Exception` to the method signature.